### DISCUSSION TITLE
How to globally replace . in branch names

### DISCUSSION THREAD

**Discussion by setchy** [renovatebot/renovate#17172] [Request Help] [answered]
We have some legacy infra automation in our CI/CD toolchain that doesn't play nicely with branches that contain `.`

Overriding the default `branchTopic` global rule and using the `replace` handlebar helper to replace `.` with `-` has proven mostly successful - [example rule](https://github.com/setchy/renovate-java-test/blob/main/renovate.json#L8)

However, there are a number of update scenarios, such as when updating a typical Java pom version property, which using the pom property key as the groupname and thus the branch name - example https://github.com/setchy/renovate-java-test/tree/renovate/major-5-netflix-dgs.version

Appreciate any suggestions on how these other scenarios can be replaced via configuration 🙏 

Alternatively, I've noticed in [branch-name.ts](https://github.com/renovatebot/renovate/blob/main/lib/workers/repository/updates/branch-name.ts#L42) it uses the [slugify package](https://github.com/simov/slugify).  Current behavior is to lower-case and replace spaces with `-`.  Would there be interest in either:
- updating the slugify function call to use `strict` mode - which would strip any special characters
- creating a new optional configuration (something like `groupSlugRemove`) which accepts a regex and when set is used to set [slugify remove](https://github.com/simov/slugify#remove)

Or similarly, allowing an optional configuration (something like `branchNameReplace`) which accepts a regex to add to the [cleanBranchName function](https://github.com/renovatebot/renovate/blob/719ab9be4a4f3482c9701f9595e513f1e12f1ce9/lib/workers/repository/updates/branch-name.ts#L22)

**Comment by rarkins** [comment]
I'm ok with adding `strict` to the current call

**Reply by viceice** [reply]
i think that will cause a lot branch renames in hosted app! So should be a breaking.

**Reply by rarkins** [reply]
Unfortunately breaking or not doesn't help the hosted app. So the second option would be something like strictBranchSlugify where we apply strict mode to everything excluding the branchPrefix?

**Reply by setchy** [reply]
@rarkins @viceice - draft PR https://github.com/renovatebot/renovate/pull/17216

**Comment by setchy** [comment, marked as answer]
Implemented in #17216
